### PR TITLE
track usage of older feature steps

### PR DESCRIPTION
**Pull request by CaseyFaist**
This PR uses the `mcount` function provided by the standard library to count how often the older legacy steps are used at build time

### CODE REVIEW

File: bin/steps/geo-libs [hunk 1]
```diff
@@ -22,6 +22,7 @@ source "$BIN_DIR/utils"
 
 # If GDAL exists within requirements, use vendored gdal.
 if [[ "$BUILD_WITH_GEO_LIBRARIES" ]]; then
+  mcount "buildvar.BUILD_WITH_GEO_LIBRARIES"
```
Comment (jmorrell) [inline bin/steps/geo-libs]: Using all caps in `mcount` is odd. Did you mean `buildvar.$BUILD_WITH_GEO_LIBRARIES` maybe?

File: bin/steps/mercurial [hunk 2]
```diff
@@ -4,6 +4,6 @@
 if [[ -f "requirements.txt" ]]; then
```
Comment (jmorrell) [inline bin/steps/mercurial]: I am curious as to why mercurial was special-cased

File: bin/steps/nltk [hunk 3]
```diff
@@ -27,6 +27,8 @@ if sp-grep -s nltk; then
         python -m nltk.downloader -d "$BUILD_DIR/.heroku/python/nltk_data" "${nltk_packages[@]}" | indent
         set_env NLTK_DATA "/app/.heroku/python/nltk_data"
 
+        mcount "buildvar.nltk_packages_definition"
+        mcount "steps.nltk"
```
Comment (jmorrell) [inline bin/steps/nltk]: Are these two going to always be the same?

Comment (jmorrell) [reply bin/steps/nltk]: You are using `steps.vendor.foo` in other cases, but `steps.nltk` here. Is that intentional?

File: bin/steps/mercurial [hunk 4]
```diff
@@ -4,6 +4,6 @@
...
 	if (grep -Fiq "hg+" requirements.txt) then
 		/app/.heroku/python/bin/pip install  mercurial | cleanup | indent
+		mcount "steps.mercurial"
```
Comment (jmorrell) [inline bin/steps/mercurial]: You are using `steps.vendor.foo` in other cases, but `steps.mercurial` here. Is that intentional?

File: bin/steps/nltk (hunk 3 above)
Comment (jmorrell) [reply bin/steps/nltk]: Oh, they might be the same, but you can make different queries `buildvar.*` and `steps.*`

File: bin/steps/geo-libs (hunk 1 above)
Comment (CaseyFaist) [reply bin/steps/geo-libs]: I started doing that for buildvars here: https://github.com/heroku/heroku-buildpack-python/blob/master/bin/steps/pip-install#L28

Is there a standard I can follow? I can update the other buildvars to be internally consistent with it if so

Comment (jmorrell) [reply bin/steps/geo-libs]: No standard, just jumped out in this diff :)

File: bin/steps/nltk (hunk 3 above)
Comment (CaseyFaist) [reply bin/steps/nltk]: "vendor" is for anything we're intentionally vendoring. We don't build our own NLTK, just download it, and so it's just a step (like mercurial). 

This is so I can aggregate on multiple dimensions, such as get all `steps`, get all `vendored` steps, get all `buildvars`, or get exactly each one in comparison

Comment (CaseyFaist) [reply bin/steps/nltk]: I missed being consistent on this buildvar output tho - fixing

**Pull request merged**